clients/src/test/java/org/apache/kafka/common/metrics/MetricsGrpcClient.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.metrics;
+
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest;
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceResponse;
+import io.opentelemetry.proto.collector.metrics.v1.MetricsServiceGrpc;
+import io.opentelemetry.proto.metrics.v1.ResourceMetrics;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import io.grpc.ConnectivityState;
+import io.grpc.ManagedChannel;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.kafka.common.utils.Utils;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+
+import java.net.UnknownHostException;
+import java.util.concurrent.TimeUnit;
+import java.net.InetAddress;
+import java.util.List;
+
+
+public class MetricsGrpcClient implements AutoCloseable {
```

Comment:
This can be package-private

```suggestion
class MetricsGrpcClient implements AutoCloseable {
```